x/participationrewards/keeper/hooks.go

```diff
@@ -146,6 +150,120 @@ func (k *Keeper) AfterZoneCreated(ctx sdk.Context, zone *icstypes.Zone) error {
 		Data: localDenomBytes,
 	})
 
+	// channel for the host chain
+	channel, found := k.IBCKeeper.ChannelKeeper.GetChannel(ctx, "transfer", zone.TransferChannel)
+	if !found {
+		return fmt.Errorf("channel not found: %s", zone.TransferChannel)
+	}
+
+	// Create LiquidAllowedDenomProtocolData for the host zone
+	hostZoneDenom := types.LiquidAllowedDenomProtocolData{
+		ChainID:               zone.ChainId,
+		RegisteredZoneChainID: zone.ChainId,
+		IbcDenom:              utils.DeriveIbcDenom(transfertypes.PortID, channel.Counterparty.ChannelId, transfertypes.PortID, zone.TransferChannel, zone.LocalDenom),
+		QAssetDenom:           zone.LocalDenom,
+	}
+	if err := hostZoneDenom.ValidateBasic(); err != nil {
+		return err
+	}
+	hostZoneDenomBytes, err := json.Marshal(hostZoneDenom)
+	if err != nil {
+		return err
+	}
+	k.SetProtocolData(ctx, hostZoneDenom.GenerateKey(), &types.ProtocolData{
+		Type: types.ProtocolDataType_name[int32(types.ProtocolDataTypeLiquidToken)],
+		Data: hostZoneDenomBytes,
+	})
+
+	// Fetch OsmosisParamsProtocolData and create LiquidAllowedDenomProtocolData for Osmosis
+	k.IteratePrefixedProtocolDatas(ctx, types.GetPrefixProtocolDataKey(types.ProtocolDataTypeOsmosisParams), func(index int64, _ []byte, data types.ProtocolData) (stop bool) {
```

Comment:
Why are we iterating here? OsmosisParams PD has a single key (it can only exist on one place). GetProtocolData would work better here.

Reply:
But we didn't know the exact chain ID as it differs between testnet and mainnet, so iterating is needed to prevent hardcode